alchemy/src/scope.ts

```diff
@@ -130,9 +130,12 @@ export class Scope {
       const orphanIds = Array.from(
         resourceIds.filter((id) => !aliveIds.has(id)),
       );
-      const orphans = await Promise.all(
-        orphanIds.map(async (id) => (await this.state.get(id))!.output),
-      );
+      const orphans = (
+        await Promise.all(
+          orphanIds.map(async (id) => (await this.state.get(id))?.output),
```

Comment:
> Better to FATAL here since that was what caught the bug.

In which case, the `!` makes it seem safe whereas you'd probably prefer something like:

```suggestion
          orphanIds.map(async (id) => {
            const orphan = await this.state.get(id);
            
            if (!oprhan) {
              throw new Error(`Unexpected error fetching orphan ${id}`);
            }
          
            return orgphan.output;
          }),
```